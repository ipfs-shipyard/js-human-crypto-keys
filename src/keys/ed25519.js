import pki from 'node-forge/lib/pki';
import ed25519 from 'node-forge/lib/ed25519';

const defaultParams = {};

const generateKeyPair = async (params, seed) => {
    // Generate the Ed25519 key-pair using node-forge by passing the seed
    const { privateKey, publicKey } = await ed25519.generateKeyPair({ seed });

    // Convert key-pair to pem base64 encoded format
    const privateKeyPem = pki.privateKeyToPem(privateKey);
    const publicKeyPem = pki.publicKeyToPem(publicKey);

    return {
        privateKey: privateKeyPem,
        publicKey: publicKeyPem,
    };
};

export { generateKeyPair, defaultParams };
