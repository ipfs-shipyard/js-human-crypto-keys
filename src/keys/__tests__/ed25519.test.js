import { generateKeyPair } from '../ed25519';
import { mockSeed } from './mocks';

describe('generateKeyPair', () => {
    it('should generate key pair succesfully', async () => {
        const keyPair = await generateKeyPair({}, mockSeed);

        expect(keyPair).toMatchSnapshot();
    });
});
