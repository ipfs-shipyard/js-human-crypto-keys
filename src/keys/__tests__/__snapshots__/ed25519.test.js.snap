// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateKeyPair should generate key pair succesfully 1`] = `
Object {
  "privateKey": Object {
    "seed": Uint8Array [
      135,
      168,
      113,
      85,
      45,
      10,
      196,
      74,
      217,
      242,
      221,
      50,
      120,
      123,
      235,
      32,
      16,
      51,
      245,
      214,
      73,
      234,
      212,
      222,
      218,
      193,
      93,
      120,
      119,
      44,
      229,
      246,
    ],
  },
  "publicKey": Object {
    "bytes": Uint8Array [
      138,
      242,
      245,
      223,
      68,
      42,
      243,
      142,
      53,
      114,
      100,
      113,
      46,
      26,
      80,
      157,
      255,
      217,
      116,
      93,
      109,
      73,
      16,
      181,
      123,
      225,
      231,
      127,
      98,
      90,
      142,
      128,
    ],
  },
}
`;
